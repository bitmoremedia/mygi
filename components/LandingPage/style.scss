@import '../../css/variables';

.site-landing-page {
  padding-top: 10px;
  @include respond-to(tablet) {
    padding-top: 30px;
  }
}
.site-landing-page__heading {
  font-size: 3rem;
  @include respond-to(tablet) {
    font-size: 4.6rem;
  }
}
.site-landing-page__app-screen-grab-img {
  width: 200px;
  margin-right: 10px;
}
.site-landing-page__gi-link {
  margin-top: 15px;
}
