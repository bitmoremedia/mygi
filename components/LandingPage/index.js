import React from 'react'
import Helmet from 'react-helmet'

import PageLink from '../common/PageLink'

import './style.scss'
import AppFoodListImg1 from './app-food-list.png'
import AppFoodListImg2 from './app-food-category.png'
import AppFoodListImg3 from './app-favourites.png'

// gatsby global workaround
let config = {}
try {
  const gatsbyConfig = require('config') // eslint-disable-line
  config = gatsbyConfig.config
} catch (e) {
  // do nothing as this is part of our gatsby mega hack
}

const LandingPage = () =>
  <div className="site-landing-page">
    <Helmet
      title={`${config.siteTitle}`}
      meta={[
        {
          name: `${config.siteTitle}`,
          content: `${config.siteDescription}`,
        },
        { property: 'og:title', content: `${config.siteTitle}` },
        { property: 'og:description', content: `${config.siteDescription}` },
        { property: 'og:image', content: `${config.appIcon}` },
        { property: 'og:url', content: `${config.baseUrl}${config.linkPrefix}` },
        { property: 'og:site_name', content: `${config.siteTitle}` },
      ]}
    />
    <h1 className="site-landing-page__heading">Welcome to MyGi</h1>
    <p>The companion site for the <b>MyGi</b> health App coming soon to iOS and Android</p>
    <div>
      <img className="site-landing-page__app-screen-grab-img" src={AppFoodListImg1} alt="Mobile App Screen Shot 1" />
      <img className="site-landing-page__app-screen-grab-img" src={AppFoodListImg2} alt="Mobile App Screen Shot 2" />
      <img className="site-landing-page__app-screen-grab-img" src={AppFoodListImg3} alt="Mobile App Screen Shot 3" />
    </div>
    <div className="site-landing-page__gi-link" >
      <PageLink to={'/glycemic-index/'}>
        <button>Glycemic Index Food List</button>
      </PageLink>
    </div>
  </div>

export default LandingPage
