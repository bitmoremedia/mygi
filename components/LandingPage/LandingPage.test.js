import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import LandingPage from './index'

test('component renders without errors', () => {
  const wrapper = mount(
    <LandingPage />,
  )
  expect(wrapper.exists()).toBe(true)
})

test('component output as expected', () => {
  const component = renderer.create(
    <LandingPage />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
