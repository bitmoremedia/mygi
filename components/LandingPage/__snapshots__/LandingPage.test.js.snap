// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component output as expected 1`] = `
<div
  className="site-landing-page"
>
  <h1
    className="site-landing-page__heading"
  >
    Welcome to MyGi
  </h1>
  <p>
    The companion site for the 
    <b>
      MyGi
    </b>
     health App coming soon to iOS and Android
  </p>
  <div>
    <img
      alt="Mobile App Screen Shot 1"
      className="site-landing-page__app-screen-grab-img"
      src="app-food-list.png"
    />
    <img
      alt="Mobile App Screen Shot 2"
      className="site-landing-page__app-screen-grab-img"
      src="app-food-category.png"
    />
    <img
      alt="Mobile App Screen Shot 3"
      className="site-landing-page__app-screen-grab-img"
      src="app-favourites.png"
    />
  </div>
  <div
    className="site-landing-page__gi-link"
  >
    <a
      className=""
      onClick={[Function]}
      style={Object {}}
    >
      <button>
        Glycemic Index Food List
      </button>
    </a>
  </div>
</div>
`;
